app/frontend/components/domains/step-code/part-3/form-section/hvac.tsx

```diff
@@ -0,0 +1,332 @@
+import { Button, Heading, Radio, RadioGroup, Stack, StackProps, Text } from "@chakra-ui/react"
+import { observer } from "mobx-react-lite"
+import { path } from "ramda"
+import React, { useEffect, useMemo } from "react"
+import { FormProvider, useController, useForm, useFormContext } from "react-hook-form"
+import { useTranslation } from "react-i18next"
+import { useLocation, useNavigate } from "react-router-dom"
+import { usePart3StepCode } from "../../../../../hooks/resources/use-part-3-step-code"
+import {
+  ECoolingSystemPlant,
+  ECoolingSystemType,
+  EDHWSystemType,
+  EHeatingSystemPlant,
+  EHeatingSystemType,
+} from "../../../../../types/enums"
+import { TextFormControl } from "../../../../shared/form/input-form-control"
+import { HStack } from "../../part-9/checklist/pdf-content/shared/h-stack"
+
+const i18nPrefix = "stepCode.part3.hvac"
+
+interface IStepcodeHvacFormProps {
+  heatingSystemPlant: EHeatingSystemPlant | null
+  heatingSystemType: EHeatingSystemType | null
+  heatingSystemTypeDescription?: string | null
+  heatingSystemPlantDescription?: string | null
+  coolingSystemPlant: ECoolingSystemPlant | null
+  coolingSystemType: ECoolingSystemType | null
+  coolingSystemTypeDescription?: string | null
+  coolingSystemPlantDescription?: string | null
+  dhwSystemType: EDHWSystemType | null
+  dhwSystemDescription?: string | null
+}
+function initializeFormValues(formValues?: IStepcodeHvacFormProps): IStepcodeHvacFormProps {
+  return {
+    heatingSystemPlant: formValues?.heatingSystemPlant || null,
+    heatingSystemType: formValues?.heatingSystemType || null,
+    heatingSystemTypeDescription: formValues?.heatingSystemTypeDescription || null,
+    heatingSystemPlantDescription: formValues?.heatingSystemPlantDescription || null,
+    coolingSystemPlant: formValues?.coolingSystemPlant || null,
+    coolingSystemType: formValues?.coolingSystemType || null,
+    coolingSystemTypeDescription: formValues?.coolingSystemTypeDescription || null,
+    coolingSystemPlantDescription: formValues?.coolingSystemPlantDescription || null,
+    dhwSystemType: formValues?.dhwSystemType || null,
+    dhwSystemDescription: formValues?.dhwSystemDescription || null,
+  }
+}
+
+export const HVAC = observer(() => {
+  const { t } = useTranslation()
+  const { checklist } = usePart3StepCode()
+  const navigate = useNavigate()
+  const location = useLocation()
+  const formMethods = useForm<IStepcodeHvacFormProps>({
+    mode: "onSubmit",
+    defaultValues: initializeFormValues(checklist),
+  })
+  const { handleSubmit } = formMethods
+
```

Comment:
better to use `Object.values(ESomeEnum)` here so if a new value is added, it is included (also cleaner code)